Examples/E3/CargoFactory.cs

```diff
@@ -0,0 +1,103 @@
+﻿namespace Examples.E3
+{
+    public interface ICargo
+    {
+        public void Book();
+    }
+    public struct CargoDeliveryLocations
+    {
+        public string Origin;
+        public string Destination;
+    }
+
+
+    #region AirCargo
+    // Im Gonna Use SingleTone Design
+    // For Instance OBJ I Used Lazy<>() Cause Its:  1-Thread Safe   2-Return Created Object Not new obj    3-Create Obj When We Need Them
+    public sealed class AirCargo : ICargo
+    {
+        private AirCargo()
+        {
+        }
+
+        private static readonly Lazy<AirCargo> instance = new Lazy<AirCargo>(() => new AirCargo());
+
+        public static AirCargo Instance => instance.Value;
+
+        public void Book()
+        {
+            Console.WriteLine("Booking via Air Cargo....");
+        }
+    }
+
+    #endregion
+
+    #region ShipCargo
+    public class ShipCargo : ICargo
+    {
+        private readonly CargoDeliveryLocations _locations;
+
+        // Using Struct For Constructor 
+        public ShipCargo(CargoDeliveryLocations locations)
+        {
+            _locations = locations;
+        }
+        public void Book()
+        {
+            if (_locations.Origin is null || _locations.Destination is null)
+            {
+                Console.WriteLine("Please Specify Locations First");
+                return;
+            }
+
+            Console.WriteLine($"Booking Via Ship Cargo From {_locations.Origin} To {_locations.Destination}");
+        }
+    }
+    #endregion
+
+    #region TrainCargo
+
+    // Considering Init Method Because it is a Part of Task
+    public class TrainCargo : ICargo
+    {
+        private bool IsInitialized = false;
```

Comment:
field must be camel-case. _isInitialized

Reply:
> field must be camel-case. _isInitialized

Yes, first it was a private prop and i decided to turn it into a normal peivate variable and i forgot to make it camelcase, iwill fix it👌🏻